src/graphene.cpp

```diff
@@ -475,49 +475,45 @@ bool CGrapheneBlock::process(CNode *pfrom,
         for (auto &kv : orphanpool.mapOrphanTransactions)
         {
             uint64_t cheapHash = kv.first.GetCheapHash();
-
-            if (mapPartialTxHash.count(cheapHash)) // Check for collisions
-                collision = true;
-
-            mapPartialTxHash[cheapHash] = kv.first;
+            auto ir = mapPartialTxHash.insert(std::make_pair(cheapHash, kv.first));
+            if (!ir.second)
+                collision = true; // insert returns false if no insertion
         }
 
         // We don't have to keep the lock on mempool.cs here to do mempool.queryHashes
         // but we take the lock anyway so we don't have to re-lock again later.
         LOCK(cs_xval);
-        mempool.queryHashes(memPoolHashes);
-
-        for (const uint256 &hash : memPoolHashes)
+        if (!collision)
         {
-            uint64_t cheapHash = hash.GetCheapHash();
+            mempool.queryHashes(memPoolHashes);
 
-            if (mapPartialTxHash.count(cheapHash)) // Check for collisions
-                collision = true;
+            for (const uint256 &hash : memPoolHashes)
+            {
+                uint64_t cheapHash = hash.GetCheapHash();
 
-            mapPartialTxHash[cheapHash] = hash;
+                auto ir = mapPartialTxHash.insert(std::make_pair(cheapHash, hash));
+                if (!ir.second)
+                    collision = true;
+            }
         }
 
         // Add full transactions included in the block
-        for (auto &tx : vAdditionalTxs)
-        {
-            const uint256 &hash = tx->GetHash();
-            uint64_t cheapHash = hash.GetCheapHash();
-
-            if (mapPartialTxHash.count(cheapHash)) // Check for collisions
-                collision = true;
+        if (!collision)
+            for (auto &tx : vAdditionalTxs)
```

Comment:
just for code readability would be good to see a parenthesis  here

+if (!collision)
+{
...
+}